x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -390,6 +390,19 @@ func (a *Aggregator) makeTransactionAggregationKey(event model.APMEvent, interva
 
 // makeMetricset makes a metricset event from key, counts, and values, with timestamp ts.
 func makeMetricset(key transactionAggregationKey, totalCount int64, counts []int64, values []float64) model.APMEvent {
+	eventSuccessCount := model.SummaryMetric{
+		Count: 0,
+		Sum:   0,
+	}
+	switch key.eventOutcome {
+	case "success":
+		eventSuccessCount.Count = totalCount
+		eventSuccessCount.Sum = float64(totalCount)
```

Comment:
Is it necessary to use `float64()` here?

Reply:
Unfortunately since we are reusing SummaryMetric here, the Sum is a float64 and it will not compile when it is assigned an int64
```
# github.com/elastic/apm-server/x-pack/apm-server/aggregation/txmetrics
x-pack/apm-server/aggregation/txmetrics/aggregator.go:400:27: cannot use totalCount (variable of type int64) as type float64 in assignment
make: *** [Makefile:61: build/apm-server-linux-amd64] Error 2
```